pypdf/_writer.py

```diff
@@ -1291,6 +1391,100 @@ def write(self, stream: Union[Path, StrByteType]) -> Tuple[bool, IO[Any]]:
 
         return my_file, stream
 
+    def list_objects_in_increment(self) -> List[IndirectObject]:
+        """
+        For debug / analysis
+        Provides the list of new/modified objects that will be written
+        in the increment
+        Deleted Objects will not be freeed but will become orphans
+
+        Returns:
+            List of (new / modified) IndirectObjects
+        """
+        return [
+            cast(IndirectObject, self._objects[i]).indirect_reference
+            for i in range(len(self._objects))
+            if (
+                self._objects[i] is not None
+                and (
+                    i >= len(self._original_hash)
+                    or cast(PdfObject, self._objects[i]).hash_bin()
+                    != self._original_hash[i]
+                )
+            )
+        ]
+
+    def _write_increment(self, stream: StreamType) -> None:
+        object_positions = {}
+        object_blocks = []
+        current_start = -1
+        current_stop = -2
+        for i, obj in enumerate(self._objects):
+            if self._objects[i] is not None and (
+                i >= len(self._original_hash)
+                or cast(PdfObject, self._objects[i]).hash_bin()
+                != self._original_hash[i]
+            ):
+                idnum = i + 1
+                assert isinstance(obj, PdfObject)  # mypy
+                # first write new/modified object
+                object_positions[idnum] = stream.tell()
+                stream.write(f"{idnum} 0 obj\n".encode())
+                """ encryption is not operational
```

Comment:
Is there something to do about this before the merge?

Reply:
no: I don't think we can encrypt an incremental file : this would mean half of the file would be cyphered (new objects) and half not (existing copied objects) : I've seen no tagging to do that. as I made this function duplicating/adjusting the standard writing I left a comment about it.